File: src/main/java/com/github/bjoernpetersen/jmusicbot/config/ConfigChecker.java
package com.github.bjoernpetersen.jmusicbot.config;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

@FunctionalInterface
public interface ConfigChecker {

  /**
   * Checks whether the given value is valid.
   *
   * @param value a config value
   * @return an warning message, or null
   */
  @Nullable
  String check(@Nonnull String value);
}

File: src/main/kotlin/com/github/bjoernpetersen/musicbot/spi/config/ConfigChecker.kt
package com.github.bjoernpetersen.musicbot.spi.config

/**
 * Checks whether the given value is valid.
 *
 * Note that, if you set a default value for the config entry, the input value for the checker will
 * never be null (or blank, if it's a String).
 *
 * @return a warning message, or null
 */
typealias ConfigChecker<T> = (T?) -> String?


